src/thing_creature.c

```diff
@@ -1242,18 +1255,204 @@ void process_thing_spell_teleport_effects(struct Thing *thing, struct CastedSpel
         {
             const struct Coord3d* newpos = NULL;
             {
-                const struct Thing* lairtng = thing_get(cctrl->lairtng_idx);
-                if (thing_is_object(lairtng)) {
-                    newpos = &lairtng->mappos;
+                switch(TeleDest)
+                {
+                    case 0:
+                    {
+                        desttng = thing_get(cctrl->lairtng_idx);
+                        break;
+                    }
+                    case 1:
+                    {
+                        rkind = RoK_TREASURE;
```

Comment:
You could save a lot of code here by removing all the cases with a room. So, case1, case2, etc.
Then have a
default:
{
    rkind=(zoom_key_room_order[TeleDest]);
}

This also makes this function more robust to other changes.

Reply:
> 
> 
> You've skipped the last zoom key, 'zoom to annoyed creature', is that on purpose?

You think that's worth including? I never thought that's a likely place someone would want to teleport a creature to.

Reply:
It's not really an important teleport location, but I think it would be nice for consistency sake. If you have 'press a zoom key in possession and teleport to go there', that's a lot nicer than 'press all but one of the zoom keys in possession to go there.'

Reply:
> 
> 
> You could save a lot of code here by removing all the cases with a room. So, case1, case2, etc.
> Then have a
> default:
> {
> rkind=(zoom_key_room_order[TeleDest]);
> }
> 
> This also makes this function more robust to other changes.

I tried that, but it appears to have broken this feature; for some reason I can't teleport to the Treasure Room any more. It does work for the Torture Chamber, but I haven't tested any other room.

Reply:
Send me the code that does not work, I'll have a look.
Did pressing T teleport you to the entrance by any chance?